src/test/java/com/getindata/connectors/http/internal/table/lookup/JavaNetHttpPollingClientWithWireTest.java

```diff
@@ -0,0 +1,153 @@
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.ExecutionOptions;
+import org.apache.flink.streaming.api.CheckpointingMode;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.getindata.connectors.http.internal.HeaderPreprocessor;
+import com.getindata.connectors.http.internal.utils.HttpHeaderUtils;
+import static com.getindata.connectors.http.TestHelper.readTestFile;
+import static com.getindata.connectors.http.internal.table.lookup.HttpLookupConnectorOptions.*;
+
+public class JavaNetHttpPollingClientWithWireTest {
+    private static final String BASE_URL = "http://localhost.com";
+
+    private static final String SAMPLES_FOLDER = "/auth/";
+    private static final int SERVER_PORT = 9090;
+
+    private static final int HTTPS_SERVER_PORT = 8443;
+
+    private static final String SERVER_KEYSTORE_PATH =
+            "src/test/resources/security/certs/serverKeyStore.jks";
+
+    private static final String SERVER_TRUSTSTORE_PATH =
+            "src/test/resources/security/certs/serverTrustStore.jks";
+
+    private static final String ENDPOINT = "/auth";
+    public static final String BEARER_REQUEST = "Bearer BBB";
+
+    private WireMockServer wireMockServer;
+    @SuppressWarnings("unchecked")
+    @BeforeEach
+    public void setup() {
+
+        File keyStoreFile = new File(SERVER_KEYSTORE_PATH);
+        File trustStoreFile = new File(SERVER_TRUSTSTORE_PATH);
+
+        wireMockServer = new WireMockServer(
+                WireMockConfiguration.wireMockConfig()
+                        .port(SERVER_PORT)
+                        .httpsPort(HTTPS_SERVER_PORT)
+                        .keystorePath(keyStoreFile.getAbsolutePath())
+                        .keystorePassword("password")
+                        .keyManagerPassword("password")
+                        .needClientAuth(true)
+                        .trustStorePath(trustStoreFile.getAbsolutePath())
+                        .trustStorePassword("password")
+                        .extensions(JsonTransform.class)
+        );
+        wireMockServer.start();
+
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        env.setRestartStrategy(RestartStrategies.noRestart());
+        Configuration config = new Configuration();
+        config.set(ExecutionOptions.RUNTIME_MODE, RuntimeExecutionMode.STREAMING);
+        env.configure(config, getClass().getClassLoader());
+        env.enableCheckpointing(1000, CheckpointingMode.EXACTLY_ONCE);
+    }
+
+    @AfterEach
+    public void tearDown() {
+        wireMockServer.stop();
+    }
+
+
+    @Test
+    public void testupdateHttpRequestIfRequiredGet() {
```

Comment:
nit: Please fix test names: test**U**pdate.
Or, as Krzysztof suggested, start test names with "should". (`ShouldUpdateGetHttpRequestIfRequired`).